measurement/plugins/ip_route/measurements.py

```diff
@@ -0,0 +1,116 @@
+"""
+ip_route measurement, formerly the Traceroute measurement.
+In a similar manner to the download_test, a list of hosts can be passed, in which case the one with the lowest latency is chosen.
+The results are returned in order as:
+ - IPRouteMeasurementResult
+ - LatencyResult of the host used to create the IPRouteMeasurement
+ - The LatencyResults from determining the initial latency
+Traceroute measurement is provided by the "scapy" library. Note that this library needs route privileges to create
+raw sockets, or to have the CAP_NET_RAW capability set.
+
+"""
+import socket
+import validators
+from validators import ValidationFailure
+
+import scapy
+
+from measurement.measurements import BaseMeasurement
+from measurement.results import Error
+from measurement.plugins.ip_route.results import IPRouteMeasurementResult
+from measurement.plugins.latency.measurements import LatencyMeasurement
+
+ROUTE_ERRORS = {
+    "route-err": "iproute encountered an unknown error",
+    "route-address": "iproute failed to find socket address",
+    "route-permission": "iproute does not have permission to create a raw socket",
+}
+
+
+class IPRouteMeasurement(BaseMeasurement):
+    def __init__(self, id, hosts, route_timeout=10, count=4):
+        super(IPRouteMeasurement, self).__init__(id=id)
+
+        if len(hosts) < 1:
+            raise ValueError("At least one url must be provided.")
+        for host in hosts:
+            validated_url = validators.domain(host)
```

Comment:
I'm not sure why `url` is used here. We're measuring a route to a host, not a url.